packages/next/src/server/node-polyfill-crypto.ts

```diff
@@ -0,0 +1,12 @@
+// Polyfill crypto() in the Node.js environment
+
+if (!(global as any).crypto) {
+  function getCryptoImpl() {
+    return require('node:crypto').webcrypto
+  }
+  Object.defineProperty(global, 'crypto', {
+    get() {
+      return getCryptoImpl()
+    },
+  })
```

Comment:
This is done similarly to how we polyfill `fetch`:  https://github.com/vercel/next.js/blob/0918ebd69a23622d78e0f25b2993beea47f5a55f/packages/next/src/server/node-polyfill-fetch.ts#L51-L68

I synced on it with @Ethan-Arrowood but I cannot recall why we needed to use `Object.defineProperty`, but I think it was compatibility/perf related.

Reply:
Yeah it's performance related, loading it before it's used adds additional time during cold boot.